File: backend/server/src/main/java/com/azat4dev/demobooking/users/presentation/api/rest/authentication/entities/FullName.java
package com.azat4dev.demobooking.users.presentation.api.rest.authentication.entities;

import jakarta.validation.constraints.NotBlank;
import org.hibernate.validator.constraints.Length;

public record FullName(
    @NotBlank @Length(min = 1, max = 255) String firstName,
    @NotBlank @Length(min = 1, max = 255) String lastName
) {
}

File: web-client/src/domain/auth/CurrentSession/Session/FullName.ts
import FirstName from "./FirstName";
import LastName from "./LastName";

class FullName {
    public constructor(
        public readonly firstName: FirstName,
        public readonly lastName: LastName
    ) {
    }

    public toString(): string {
        return `${this.firstName.value} ${this.lastName.value}`;
    }

    public getInitials = (): string => {
        return `${this.firstName.value[0]}${this.lastName.value[0]}`;
    }
}

export default FullName;
